dotnet/src/Connectors/Connectors.OpenAI/TextToImage/OpenAITextToImageExecutionSettings.cs

```diff
@@ -0,0 +1,159 @@
+    public OpenAITextToImageExecutionSettings()
+    {
+        this.Width = DefaultWidth;
+        this.Height = DefaultHeight;
+    }
+    /// <summary>
+    /// Width of the generated image.
+    /// </summary>
+    public int Width
+    {
+        get => this._width;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._width = value;
+        }
+    }
+
+    /// <summary>
+    /// The quality of the image that will be generated.
+    /// `hd` creates images with finer details and greater consistency across the image.
+    /// This param is only supported for dall-e-3.
+    /// </summary>
+    public string? Quality
+    {
+        get => this._quality;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._quality = value;
+        }
+    }
+
+    /// <summary>
+    /// The number of images to generate. Must be between 1 and 10.
+    /// For dall-e-3, only ImageCount = 1 is supported.
+    /// </summary>
+    public int? ImageCount
+    {
+        get => this._imageCount;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._imageCount = value;
+        }
+    }
+
+    /// <summary>
+    /// The style of the generated images. Must be one of vivid or natural.
+    /// Vivid causes the model to lean towards generating hyper-real and dramatic images.
+    /// Natural causes the model to produce more natural, less hyper-real looking images.
+    /// This param is only supported for dall-e-3.
+    /// </summary>
+    public string? Style
+    {
+        get => this._style;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._style = value;
+        }
+    }
+
+    /// <summary>
+    /// Height of the generated image.
+    /// </summary>
+    public int Height
+    {
+        get => this._height;
+
+        set
+        {
+            this.ThrowIfFrozen();
+            this._height = value;
+        }
+    }
+
+    /// <inheritdoc/>
+    public override void Freeze()
+    {
+        if (this.IsFrozen)
+        {
+            return;
+        }
+
+        base.Freeze();
+    }
+
+    /// <inheritdoc/>
+    public override PromptExecutionSettings Clone()
+    {
+        return new OpenAITextToImageExecutionSettings()
+        {
+            ModelId = this.ModelId,
+            ExtensionData = this.ExtensionData is not null ? new Dictionary<string, object>(this.ExtensionData) : null,
+            Width = this.Width,
+            Height = this.Height,
+        };
+    }
+
+    /// <summary>
+    /// Create a new settings object with the values from another settings object.
+    /// </summary>
+    /// <param name="executionSettings">Template configuration</param>
+    /// <param name="defaultMaxTokens">Default max tokens</param>
+    /// <returns>An instance of OpenAIPromptExecutionSettings</returns>
+    public static OpenAITextToImageExecutionSettings FromExecutionSettings(PromptExecutionSettings? executionSettings, int? defaultMaxTokens = null)
```

Comment:
seemed the `defaultMaxTokens` parameter not used

Reply:
Removed